File: win/Handbrake/frmUpdate.vb
Imports System.IO

Public Class frmUpdate

    Private Sub frmUpdate_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If My.Settings.StartupUpdate = 1 Then
            lbl_startupStatus.Text = "On"
        Else
            lbl_startupStatus.Text = "Off"
        End If

        Version.Text = My.Settings.HandbrakeGUIVersion
        cliVersion.Text = My.Settings.HandbrakeCLIVersion
    End Sub

    Private Sub Dest_browse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Dest_browse.Click
        Dim file_path As String = Application.StartupPath
        Try
            ' Download the update file
            ' open the file for reading and read the first 2 lines for GUI and CLI versions
            Dim wc As New System.Net.WebClient()
            wc.DownloadFile("http://download.m0k.org/handbrake/windows/update.txt", file_path & "\update.txt")
            wc.Dispose()
            Dim versionStream As StreamReader = File.OpenText(file_path & "\update.txt")
            Dim windowsGUI As String = versionStream.ReadLine()
            Dim windowsCLI As String = versionStream.ReadLine()
            versionStream.Close()

            ' Set the Latest Text label to the first line of the file
            lbl_latest.Text = windowsGUI
            lbl_encoderVersion.Text = windowsCLI

            ' If the version is now the same as the one shown here, Display the update label
            If (windowsGUI <> My.Settings.HandbrakeGUIVersion Or windowsCLI <> My.Settings.HandbrakeCLIVersion) Then
                MessageBox.Show("A new version is available. Please visit the project website to download the update.", "Status", MessageBoxButtons.OK, MessageBoxIcon.Asterisk)
            End If


        Catch ex As Exception
            ' Handdle any errors that may occur
            MessageBox.Show("Unable to check for updates. The server may be unavailible at the moment. Please try again later!", "Error", MessageBoxButtons.OK, MessageBoxIcon.Hand)
        End Try


    End Sub

    Private Sub btn_close_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btn_close.Click
        Me.Close()
    End Sub

End Class

File: win/C#/frmUpdate.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using System.Net;

namespace Handbrake
{
    public partial class frmUpdate : Form
    {
        public frmUpdate()
        {
            InitializeComponent();
        }

        private void btn_update_click(object sender, EventArgs e)
        {
            // Try to download the contents of the update file. If sucessful then split the 2 lines and do an update check/
            try
            {
                String updateFile = Properties.Settings.Default.updateFile;
                WebClient client = new WebClient();
                String data = client.DownloadString(updateFile);
                String[] versionData = data.Split('\n');

                lbl_GuiVersion.Text = versionData[0];
                lbl_cliVersion.Text = versionData[1];

                if ((versionData[0] != Properties.Settings.Default.GuiVersion) || (versionData[1] != Properties.Settings.Default.CliVersion))
                {
                    MessageBox.Show("A new version is available. Please visit the project website to download the update.", "Status", MessageBoxButtons.OK, MessageBoxIcon.Asterisk);
                }
            }
            //else fail displaying an error message.
            catch (Exception)
            {

                MessageBox.Show("Unable to check for new version at this time. Please try again later.", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        private void btn_close_Click(object sender, EventArgs e)
        {
            this.Close(); // Close the window
        }

        private void frmUpdate_Load(object sender, EventArgs e)
        {
            // Set the labels with their correct values.
            Version.Text = Properties.Settings.Default.GuiVersion;
            cliVersion.Text = Properties.Settings.Default.CliVersion;

            // Display On/Off depending on the updateStatus setting
            string updateStatus = Properties.Settings.Default.updateStatus;

            if(updateStatus == "1"){
                lbl_startupStatus.Text = "On";
            }else{
                lbl_startupStatus.Text = "Off";
            }
        }
    }
}